modules/renter/contractor/editor.go

```diff
@@ -110,109 +104,55 @@ func (he *hostEditor) Upload(data []byte) (_ crypto.Hash, err error) {
 	if he.invalid {
 		return crypto.Hash{}, errInvalidEditor
 	}
-	contract, sectorRoot, err := he.editor.Upload(data)
+
+	// Perform the upload.
+	_, sectorRoot, err := he.editor.Upload(data)
 	if err != nil {
 		return crypto.Hash{}, err
 	}
-	he.contractor.mu.Lock()
-	he.contractor.contracts[contract.ID] = contract
-	he.contractor.persist.update(updateUploadRevision{
-		NewRevisionTxn:     contract.LastRevisionTxn,
-		NewSectorRoot:      sectorRoot,
-		NewSectorIndex:     len(contract.MerkleRoots) - 1,
-		NewUploadSpending:  contract.UploadSpending,
-		NewStorageSpending: contract.StorageSpending,
-	})
-	he.contractor.mu.Unlock()
-	he.contract = contract
-
 	return sectorRoot, nil
 }
 
-// Delete negotiates a revision that removes a sector from a file contract.
-func (he *hostEditor) Delete(root crypto.Hash) (err error) {
-	he.mu.Lock()
-	defer he.mu.Unlock()
-	if he.invalid {
-		return errInvalidEditor
-	}
-	contract, err := he.editor.Delete(root)
-	if err != nil {
-		return err
-	}
-
-	he.contractor.mu.Lock()
-	he.contractor.contracts[contract.ID] = contract
-	he.contractor.saveSync()
-	he.contractor.mu.Unlock()
-	he.contract = contract
-
-	return nil
-}
-
-// Modify negotiates a revision that edits a sector in a file contract.
-func (he *hostEditor) Modify(oldRoot, newRoot crypto.Hash, offset uint64, newData []byte) (err error) {
-	he.mu.Lock()
-	defer he.mu.Unlock()
-	if he.invalid {
-		return errInvalidEditor
-	}
-	contract, err := he.editor.Modify(oldRoot, newRoot, offset, newData)
-	if err != nil {
-		return err
-	}
-	he.contractor.mu.Lock()
-	he.contractor.contracts[contract.ID] = contract
-	he.contractor.saveSync()
-	he.contractor.mu.Unlock()
-	he.contract = contract
-
-	return nil
-}
-
 // Editor returns a Editor object that can be used to upload, modify, and
 // delete sectors on a host.
 func (c *Contractor) Editor(id types.FileContractID, cancel <-chan struct{}) (_ Editor, err error) {
 	id = c.ResolveID(id)
 	c.mu.RLock()
 	cachedEditor, haveEditor := c.editors[id]
 	height := c.blockHeight
-	contract, haveContract := c.contracts[id]
 	renewing := c.renewing[id]
 	c.mu.RUnlock()
 
 	if renewing {
+		// Cannot use the editor if the contract is being renewed.
```

Comment:
This is another check that might no longer be necessary if the contractset is now providing appropriate safety.